operator/internal/manifests/storage/configure.go

```diff
@@ -186,29 +186,58 @@ func ensureObjectStoreCredentials(p *corev1.PodSpec, opts Options) corev1.PodSpe
 			},
 		}
 	case lokiv1.ObjectStorageSecretS3:
-		storeEnvVars = []corev1.EnvVar{
-			{
-				Name: EnvAWSAccessKeyID,
-				ValueFrom: &corev1.EnvVarSource{
-					SecretKeyRef: &corev1.SecretKeySelector{
-						LocalObjectReference: corev1.LocalObjectReference{
-							Name: secretName,
+		sts := opts.S3 != nil && len(opts.S3.RoleArn) != 0
```

Comment:
WDYT pushing this condition into a method of S3, e.g. `opts.IsSTS()` (please feel free to improve naming). We can make use of it in the `loki-config.yaml` template and improve readability.

Reply:
I think I went a bit in this direction but maybe not what you had in mind. I'll not mark this as resolved and I want to have this into consideration when the code is finished as to see if I can improve it